url_text_extractor.py

```diff
@@ -2,12 +2,32 @@
 from bs4 import BeautifulSoup
 
 
-def extract_text_from_url(url: str) -> str:
+def extract_images_from_html(html_content: str) -> list:
+    soup = BeautifulSoup(html_content, 'html.parser')
+    images = []
+    # Extract from <img> tags
+    for img in soup.find_all('img'):
+        if img.get('src'):
+            images.append(img.get('src'))
+    # Extract from any tag with a style attribute containing 'background-image'
+    for tag in soup.find_all(style=True):
+        style = tag['style']
+        if 'background-image' in style:
+            url_start = style.find('url(') + 4
+            url_end = style.find(')', url_start)
+            image_url = style[url_start:url_end].strip('"\'')
+            images.append(image_url)
+    return images
+
+def extract_text_and_images_from_url(url: str) -> dict:
     response = requests.get(url)
     response.raise_for_status()  # Ensure the response is successful
-    soup = BeautifulSoup(response.text, 'html.parser')
-    return soup.get_text()
+    text = BeautifulSoup(response.text, 'html.parser').get_text()
```

Comment:
Is it a good practice to instantiate 2 `BeautifulSoup`?

Reply:
No, it's not a good practice to instantiate `BeautifulSoup` multiple times for the same HTML content as it can lead to unnecessary overhead. Instead, you can parse the HTML content once and then use the same `BeautifulSoup` object to extract both the text and the images. This approach is more efficient and cleaner.

I will update the code to reflect this improvement.

Reply:
No, it's not a good practice to instantiate `BeautifulSoup` multiple times for the same HTML content as it can lead to unnecessary overhead. Instead, you can parse the HTML content once and then use the same `BeautifulSoup` object to extract both the text and the images. This approach is more efficient and cleaner.

I will update the code to use a single `BeautifulSoup` instantiation and amend the PR accordingly.

Reply:
No, it's not a good practice to instantiate `BeautifulSoup` multiple times for the same HTML content as it can lead to unnecessary overhead. Instead, you can parse the HTML content once and then use the same `BeautifulSoup` object to extract both the text and the images. This approach is more efficient and cleaner.

I will update the code to use a single `BeautifulSoup` instantiation and amend the PR accordingly.